protostar/starknet_gateway/gateway_facade.py

```diff
@@ -446,6 +449,33 @@ async def _create_contract_function(
         except KeyError:
             raise UnknownFunctionException(function_name) from KeyError
 
+    async def _get_resolved_contract_function(
+        self,
+        contract_address: AddressRepresentation,
+        client: Optional[Client] = None,
+    ):
+        try:
+            try:
+                contract = await Contract.from_address(
+                    address=contract_address,
+                    client=client or self._gateway_client,
+                    proxy_config=True,
+                )
+                return contract
+            except ClientError as err:
```

Comment:
variable is not needed

Reply:
?. The variable is used.

Reply:
it's returned right away

Reply:
NVM I thought you meant `err` variable for some reason.